src/puppetlabs/puppetdb/command/dlo.clj

```diff
@@ -71,31 +76,43 @@
                           (.printStackTrace exception out))
                         attempts))))
 
+(defn- strip-metadata-suffix
+  [queue-metadata-str]
+  (str/replace queue-metadata-str #"\.json$" ""))
```

Comment:
Why? That's both harder to understand and more likely to do the wrong thing (if the metadata-str doesn't have the correct suffix).